### ISSUE TITLE
When local branch diverges from the remote and you're about to force push, the warning should prompt the user enter some text instead of just pressing enter.

### ISSUE THREAD

**Issue by iltenahmet** [jesseduffield/lazygit#3468]
**Is your feature request related to a problem? Please describe.**
When the local branch has diverged from the remote and you're about to force push, there is a super useful warning that says `your branch has diverged from the remote branch, Press <esc> to cancel, or <enter> to force push.` Unfortunately, I ended up pressing `enter` out of habit too many times because it's just muscle memory at this point. We could avoid so many accidental force pushes if we had prompted to user the enter a word instead.

**Describe the solution you'd like**
The pop-up could be changed as `your branch has diverged from the remote branch, Press <esc> to cancel, or type "force" to force push.` It can be another word, or it could be something else that prevents the user from just pressing `enter` out of habit.

**Additional context**
Many large projects employ this tactic because it's too easy to press buttons without reading what they do. For example, when you're changing the visibility of a repo on Github, they ask you to type the name of the repo in addition to the warning pop-up. This ensures people don't just press `enter` out of habit without even reading what the pop-up says.

**Comment by belyakov-am** [comment]
This looks useful to me too. And I would also consider changing pop-up for `discard` action in the same way.

However, I would prefer having a config options for each action (forse push, discard, etc) and enabling/disabling them separately.

**Comment by AndrewSav** [comment]
https://github.com/jesseduffield/lazygit/issues/2009 IMO this should be configurable, if someone prefers to force push fast, they should be able to do so, but those who wants extra safety should also have that option.

**Comment by simondoesstuff** [comment]
🙏 Just did an inadvertent force push resulting in 2 weeks of a collaborator's deleted work and 3 hours finding the dangling commits to recover.

**Comment by iltenahmet** [comment]
@simondoesstuff this is exactly why I opened this issue. Based on the discussion on #2009, we are not the first ones to do so. The potentially dire consequences of accidental force pushes are far greater than the benefits of being able to force push fast. 

I agree that it should be configurable, but it makes sense to have the safer option as the default, especially because I think most people will not go through the extra step of configuring this until they make an inadvertent force push.

**Comment by socketbox** [comment]
Has the `[--force-with-lease](https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force)` flag been taken into consideration in this discussion?

**Comment by dtabuenc** [comment]
> Has the [`--force-with-lease`](https://stackoverflow.com/questions/52823692/git-push-force-with-lease-vs-force) flag been taken into consideration in this discussion?

`--force-withlease` does absolutely nothing in practice. In 99.99% of the use-cases with lazygit it will work exactly the same as just a plain `--force`, since lazygit is always fetching the remote refs in the background. Force-pushing is ridculously easy in lazygit... I can't believe the default is to just press "enter" for such a destructive operation.

**Comment by stefanhaller** [comment]
> `--force-with-lease` does absolutely nothing in practice. In 99.99% of the use-cases with lazygit it will work exactly the same as just a plain `--force`, since lazygit is always fetching the remote refs in the background.

Yes, and this has been discussed a lot of times in various issues and discussions here. Here's [an example](https://github.com/jesseduffield/lazygit/discussions/4068#discussioncomment-11336272).

The recommendation is to use `git config --global push.useForceIfIncludes true` to make `--force-with-lease` really safe. Unfortunately, lazygit can't do this, for reasons explained in the link above.

**Comment by jesseduffield** [comment]
I'm in favour of a configuration option that allows adding more friction to force pushes, but I want the default behaviour to remain as-is for the sake of all the people who have a rebase-heavy workflow which necessarily involves lots of force pushing.

I'll happily accept a PR for this.
